### PR TITLE
docs(operators): add documentation for takeWhile

### LINKED ISSUE #103: Add docs for operator : filtering - takeWhile
Add docs for operator takeWhile under group filtering 
 
Below is minimum details needed 
 
- [ ] name 
- [ ] operatorType 
- [ ] signature 
- [ ] parameters 
- [ ] short description 
- [ ] Technical description 
- [ ] Visual example  (Marble diagram) 
- [ ] Real world example (Non-framework) 
- [ ] Related Operators

### PR DESCRIPTION
**Pull request by ashwin-sureshkumar**
Close #103

### CODE REVIEW

File: src/operator-docs/filtering/takeWhile.ts [hunk 1]
```diff
@@ -1,6 +1,66 @@
 import { OperatorDoc } from '../operator.model';
 
 export const takeWhile: OperatorDoc = {
-  'name': 'takeWhile',
-  'operatorType': 'filtering'
+  name: 'takeWhile',
+  operatorType: 'filtering',
+  signature:
+    'public takeWhile(predicate: function(value: T, index: number): boolean): Observable<T>',
```
Comment (JWO719) [inline src/operator-docs/filtering/takeWhile.ts]: @btroncone suggested to remove the generic from the signature

File: src/operator-docs/filtering/takeWhile.ts [hunk 2]
```diff
@@ -1,6 +1,66 @@
...
+  shortDescription: {
+    description: `
+    Emits values emitted by the source Observable so long as each value satisfies the given predicate,
+     and then completes as soon as this predicate is not satisfied.
+    `,
+    extras: [
+      {
+        type: 'Tip',
+        text: `
+        Takes values from the source only while they pass the condition given. When the first value does not satisfy, it completes.
```
Comment (JWO719) [inline src/operator-docs/filtering/takeWhile.ts]: I think the tip is pretty redundant to the description above.

File: src/operator-docs/filtering/takeWhile.ts (hunk 1 above)
Comment (ashwin-sureshkumar) [reply src/operator-docs/filtering/takeWhile.ts]: @JWO719 @btroncone  - Generic on the return type ? Just so I am on the page.

File: src/operator-docs/filtering/takeWhile.ts [hunk 3]
```diff
@@ -1,6 +1,63 @@
...
+      code: `
+      import { fromEvent } from 'rxjs/observable/fromEvent';
+      import { takeWhile } from 'rxjs/operators';
+
+      const clicks = fromEvent(document, 'click');
+      const result = clicks.pipe(
+                        takeWhile(ev => ev.clientX > 100)
+                     );
+      result.subscribe(x => console.log(x));
+      `,
```
Comment (sumitarora) [inline src/operator-docs/filtering/takeWhile.ts]: same here sample output?

File: src/operator-docs/filtering/takeWhile.ts (hunk 1 above)
Comment (JWO719) [reply src/operator-docs/filtering/takeWhile.ts]: if i understand @btroncone at #228 correctly, yes

Comment (btroncone) [reply src/operator-docs/filtering/takeWhile.ts]: @ashwin-sureshkumar @JWO719 Yes, let's leave out in signatures per https://github.com/ReactiveX/rxjs-docs/pull/196#discussion_r157231328

**Comment by ashwin-sureshkumar** [comment]
@JWO719 @sumitarora - Done, please re-review.

**Comment by ladyleet** [comment]
yay thanks @sumitarora for merging!

**Pull request merged**